### PR TITLE
Gun Size Rebalance, New Lever-Shotgun & Plasma AOE

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request
<details>
<summary>
	Tl;dr - Rebalanced gun sizes mostly downwards. Allows more guns to be put into backpacks, such as the SWAT shotgun, grizzly, STS-30 (AK fits in a backpack, why doesn't an STS) and so-forth. Reductor still fits but has received 'extra bulk' to take up some extra space. Added the new Mares Leg lever-shotgun made out of an armstrong sprited by BoblinTheGoblin21#2680. Also removed the knockback plasma impact since it's very silly to have a hard-stun (stun = 1 and knockdown = 2 stacks stun timers) - so the Dominion rifle has had its blast-down reworked into a unique fire-AOE effect, making it much better in PvE and less gamey in PvP.
</summary>
<hr>
	
<hr>
</details>

## Changelog
:cl:
add: New Mareleg lever-action shotgun. Can be crafted using an Armstrong.
balance: Rebalances gun sizes mostly across the board. Makes more guns able to be backpacked, even if they are large.
tweak: Replaces Dominion hard-knockback firemode with a more PvE useful AOE fire option.
removes: Laser tags from having RNG knockback. Was suggested, see no reason not to since it's a silly 'one-hit' hard-stun.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/boltgun/armstrong.dm [hunk 1]
```diff
@@ -4,19 +4,39 @@
...
 	slot_flags = SLOT_BELT|SLOT_BACK
 	force = WEAPON_FORCE_PAINFUL
 	caliber = CAL_MAGNUM
 	max_shells = 11
-	price_tag = 650
+	price_tag = 600
 	init_recoil = HMG_RECOIL(0.4)
 	damage_multiplier = 1.2
 	penetration_multiplier  = 1.1
+	zoom_factor = 0.4
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/projectile/boltgun/armstrong.dm:16]: if this has a built in scope it prob shouldnt be able to take scopes

Comment (Stutternov) [reply code/modules/projectiles/guns/projectile/boltgun/armstrong.dm:16]: Fixed

**Comment by cdb-is-not-good** [comment]
Not a super big fan of making more guns fit in bags without increasing their bulk, but I'll have to take a closer look when I get a sec

**Comment by Stutternov** [comment]
> Not a super big fan of making more guns fit in bags without increasing their bulk, but I'll have to take a closer look when I get a sec

AKs have always fit into bags (including the breacher) with no extra bulk. Same for even the Reductor, which I have given extra bulk to since it’s a OP shotgun in essence. Guns that are _HUGE can never fit in bags. Guns that are _BULKY can. STSs should have been able to fit in bags without being sawn down for awhile since it’s meant to be roughly on par with an AK.

Ion rifles are huge when - they shouldn’t be. They are incredibly niche and cannot hurt anything non-electronic. (Plus if the critique of anyone is ‘oh but they instant kill my FBP/cyborg/modded human’ - ion pistols do the same damage and fit in pockets. The rifles just have more charge.) 

Kalash-bren should be the same size as the AK. It’s just a weaker AK with a higher RPM. (My fault from ages ago when I made the thing not balancing it with an AK.)

Armstrong is weak compared to other voltier; it’s .40 magnum. So, it gives it a competitive edge at least to someone obeying gun laws if it can’t be seen down any. 

Also, keep in mind, most recently added energy weapons (from the SST to others) can all fit in backpacks. Yet many times carry more strength than their counterparts.

Reason for the change is the Longarm law since we buffed mobs and have random events that can make carps, blobs, hive minds, Kaiser’s and other hazards spawn everywhere. This should allow people to carry some mid-grade weaponry but sacrifice inventory space instead of using gun cases that have a mild limit in supply or pistols that really cannot kill armored random spawn mobs at all.

File: code/modules/projectiles/guns/energy/cog.dm [hunk 2]
```diff
@@ -25,6 +25,7 @@
 	name = "\"Gear\" lasgun"
 	desc = "A Greyson Positronic design, cheap and widely produced. In the distant past - this was the main weapon of low-rank police forces, billions of copies of this gun were made. They are ubiquitous. \
 	This model has been modified by the Marshals to allow for non-lethal electrodes to be discharged as well as lasers, but at the cost of its cell-usage effifency."
+	w_class = ITEM_SIZE_NORMAL
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/cog.dm:28]: Form bulky to normal?

File: code/modules/projectiles/guns/energy/ionrifle.dm [hunk 3]
```diff
@@ -7,7 +7,7 @@
 	item_charge_meter = TRUE
 	fire_sound = 'sound/weapons/Laser.ogg'
 	origin_tech = list(TECH_COMBAT = 2, TECH_MAGNET = 4)
-	w_class = ITEM_SIZE_HUGE
+	w_class = ITEM_SIZE_BULKY
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/ionrifle.dm:10]: 100% agree with this one, ion-stuff is still underused and undervalued

File: code/modules/projectiles/guns/energy/plasma.dm [hunk 4]
```diff
@@ -22,7 +22,7 @@
 	damage_multiplier = 0.9
 	init_firemodes = list(
 		list(mode_name="destroy", mode_desc="An armor-stripping plasma round", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=14, icon="destroy", projectile_color = "#FFFFFF"),
-		list(mode_name="impact", mode_desc="A relatively light plasma round that knocks people back", projectile_type=/obj/item/projectile/plasma/impact, fire_sound='sound/effects/supermatter.ogg', fire_delay=25, icon="kill", projectile_color = "#FF0000"),
+		list(mode_name="incinerate", mode_desc="A relatively heavy shot of super-heated plasma that creates a fireball on impact", projectile_type=/obj/item/projectile/plasma/aoe/heat, fire_sound='sound/effects/supermatter.ogg', charge_cost=160, fire_delay=25, icon="kill", projectile_color = "#FF0000"),
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/plasma.dm:25]: Personally requested by me, huge improvement on balance, as well as not a direct nerf meaning that its a tweak in my eyes as it has much more value in what the gun is meant for, large scale combat not stunlocking a single threat

File: code/modules/projectiles/guns/energy/railgun.dm [hunk 5]
```diff
@@ -9,6 +9,7 @@
 	fire_sound = 'sound/weapons/rail.ogg'
 	item_charge_meter = TRUE
 	w_class = ITEM_SIZE_BULKY
+	extra_bulk = 2
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/railgun.dm:12]: adding 2 bulk is the equalivant of making it have a small items more size when in modular storage.
This is fine

File: code/modules/projectiles/guns/projectile/automatic/bren.dm [hunk 6]
```diff
@@ -5,7 +5,7 @@
 	icon = 'icons/obj/guns/projectile/bren.dmi'
 	icon_state = "bren"
 	item_state = "bren"
-	w_class = ITEM_SIZE_HUGE
+	w_class = ITEM_SIZE_BULKY
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/projectile/automatic/bren.dm:8]: Personally think the bren is over-valued as a simple omni or sts tends to be best.
Im alright with reducing the size

File: code/modules/projectiles/guns/energy/cog.dm (hunk 2 above)
Comment (Stutternov) [reply code/modules/projectiles/guns/energy/cog.dm:28]: This is to make it on-par with the Mamba. I could add a folding stock to it instead if preferred.

File: code/modules/projectiles/guns/projectile/automatic/bulldog.dm [hunk 7]
```diff
@@ -4,7 +4,7 @@
 	icon = 'icons/obj/guns/projectile/bulldog.dmi'
 	icon_state = "bulldog"
 	item_state = "bulldog"
-	w_class = ITEM_SIZE_BULKY
+	w_class = ITEM_SIZE_NORMAL
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/projectile/automatic/bulldog.dm:7]: I frankly dont understand this change at all.
Its a massive buff to a descent gun that is out of circulation?
Dosnt matter to much I suppose

Comment (Stutternov) [reply code/modules/projectiles/guns/projectile/automatic/bulldog.dm:7]: Error on my end it seems, I didn’t notice till now I reduced that one further. Don’t know if I copy-pasted the wrong value or what on that one. I’ll amend that.

File: code/modules/projectiles/guns/projectile/shotgun/pump.dm [hunk 8]
```diff
@@ -56,7 +56,7 @@
 	icon = 'icons/obj/guns/projectile/sawnoff/shotgun.dmi'
 	icon_state = "shotgun"
 	item_state = "shotgun"
-	w_class = ITEM_SIZE_BULKY
+	w_class = ITEM_SIZE_NORMAL
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/projectile/shotgun/pump.dm:59]: Why would I ever saw it down again?

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]
Only changes I see needed are the shotgun stuff being un-sized changed as it messes with their value of sawn a bit to much

File: code/modules/projectiles/guns/projectile/shotgun/pump.dm (hunk 8 above)
Comment (Stutternov) [reply code/modules/projectiles/guns/projectile/shotgun/pump.dm:59]: Had discussion about it - amending this. Grizzly will be reverted to huge n’ left as is.

**Comment by Stutternov** [comment]
> Only changes I see needed are the shotgun stuff being un-sized changed as it messes with their value of sawn a bit to much

Fixed.

File: code/modules/projectiles/guns/energy/cog.dm (hunk 2 above)
Comment (Stutternov) [reply code/modules/projectiles/guns/energy/cog.dm:28]: Actually error - didn't realize I set it to normal. Amended.

**Review by Trilbyspaceclone — APPROVED** [review]
